src/test/regress/regress_gp.c

```diff
@@ -641,8 +643,23 @@ resGroupPalloc(PG_FUNCTION_ARGS)
 	if (!IsResGroupEnabled())
 		PG_RETURN_INT32(0);
 
+	if (startUpMbRemains == -1)
+	{
+		startUpMbRemains =
+			(VmemTracker_GetStartupChunks())
+			<< (VmemTracker_GetChunkSizeInBits() - BITS_IN_MB);
+	}
+
 	ResGroupGetMemInfo(&memLimit, &slotQuota, &sharedQuota);
 	size = ceilf(memLimit * ratio);
+	// At startup, the backend process is already consuming some amount of
+	// memory. In order not to complicate the logic of the tests, we take this
+	// memory into account when allocating memory for tests.
+	if (startUpMbRemains > 0)
+	{
+		size = Max(0, size - startUpMbRemains);
+		startUpMbRemains = Max(0, startUpMbRemains - size);
+	}
```

Comment:
The current code calls MemoryContextAlloc when size is 0.
When the input is startUpMbRemains = 12 and size = 4, the output is size = max(0, 4-12) = 0 and startUpMbRemains = Max(0,12-0) = 12. But the output startUpMbRemains should be 8
```suggestion
	if (startUpMbRemains >= size)
	{
		startUpMbRemains -= size;
		PG_RETURN_INT32(0);
	}

	size -= startUpMbRemains;
	startUpMbRemains = 0;
```

Reply:
done